lib/datadog/kit/events.rb

```diff
@@ -7,41 +7,62 @@ module Datadog
       #
       # @param trace [TraceOperation] Trace to attach data to.
-      # @param success [bool] Whether login operation was successful or not.
       # @param user [Hash<Symbol, String>] User information to pass to
       #   Datadog::Kit::Identity.set_user. Must contain at least :id as key.
       # @param others [Hash<String || Symbol, String>] Additional free-form
       #   event information to attach to the trace.
-      def self.track_login(trace, success:, user:, **others)
-        subtag = success ? 'success' : 'failure'
-        event = "users.login.#{subtag}"
-
-        track(event, trace, **others)
+      def self.track_login_success(trace, user:, **others)
+        track(:appsec, APPSEC_LOGIN_SUCCESS_EVENT, trace, **others)
 
         user_options = user.dup
         user_id = user.delete(:id)
 
         raise ArgumentError, 'missing required key: :user => { :id }' if user_id.nil?
 
-        if success
-          Kit::Identity.set_user(trace, id: user_id, **user_options)
-        else
-          trace.set_tag("appsec.events.users.login.#{subtag}.usr.id", user_id)
-        end
+        Kit::Identity.set_user(trace, id: user_id, **user_options)
+      end
+
+      # Attach login failure event information to the trace
+      #
+      # This method is experimental and may change in the future.
+      #
+      # @param trace [TraceOperation] Trace to attach data to.
+      # @param user_id [String] User id that attempted login
+      #   Datadog::Kit::Identity.set_user. Must contain at least :id as key.
+      # @param user_exists [bool] Whether the user id that did a login attempt exists.
+      # @param others [Hash<String || Symbol, String>] Additional free-form
+      #   event information to attach to the trace.
+      def self.track_login_failure(trace, user_id:, user_exists:, **others)
+        track(:appsec, APPSEC_LOGIN_FAILURE_EVENT, trace, **others)
+
+        raise ArgumentError, 'missing required key: :user => { :id }' if user_id.nil?
+
+        trace.set_tag('appsec.events.users.login.failure.usr.id', user_id)
+        trace.set_tag('appsec.events.users.login.failure.usr.exists', user_exists)
       end
 
       # Attach custom event information to the trace
       #
+      # This method is experimental and may change in the future.
+      #
+      # @param namespace [Symbol] Mandatory. Event namespace. Only :appsec is supported.
       # @param event [String] Mandatory. Event code.
       # @param trace [TraceOperation] Trace to attach data to.
       # @param others [Hash<Symbol, String>] Additional free-form
       #   event information to attach to the trace. Key must not
       #   be :track.
-      def self.track(event, trace, **others)
-        trace.set_tag("appsec.events.#{event}.track", 'true')
+      def self.track(namespace, event, trace, **others)
+        raise ArgumentError, "namespace cannot be #{namespace.inspect}" if namespace.to_sym != :appsec
```

Comment:
Minor: I suggest extending the error to make it a bit more self-describable.

E.g. right now if I see `ArgumentError: namespace cannot be :potato` I've learned that I did something wrong, but I'll need to go read the code or the API docs to understand what I did wrong exactly and how to fix it; whereas with `ArgumentError: namespace cannot be :potato, only :appsec is allowed` then I immediately know that I need to either switch to the allowed namespace or reconsider using this API.